src/main/java/com/lgtm/easymoney/services/impl/TransferServiceImpl.java

```diff
@@ -68,30 +70,28 @@ public ResponseEntity<TransferRsp> makeTransfer(TransferReq req) {
             createTransaction(fromUser, toUser, amount, Category.valueOf(category.toUpperCase()), desc);
     boolean success = makeTransfer(transaction);
     // payload
-    TransferRsp response = new TransferRsp();
-    response.setSuccess(success);
-    response.setCurrBalance(fromUser.getBalance());
-    List<TransactionRsp> transferRsps =
-            transactionService.generateListResponseFromTransactions(List.of(transaction));
-    response.setTransfers(transferRsps);
-    if (success) {
-      return ResponseEntity.status(HttpStatus.OK).body(response);
+    if (!success) {
+      transaction.setStatus(TransactionStatus.TRANS_FAILED);
```

Comment:
can we refactor `executeTransaction` to save the transaction with proper status even if it fails(potentially throw exceptions)? In this way we benefit from
1. upper level service don't need to worry about this
2. easier to trace bugs
3. cleaner code

Reply:
Currently don't change after offline discussion